creusot-contracts/src/std/pointer.rs

```diff
@@ -0,0 +1,92 @@
+use crate::*;
+
+// We conservatively model raw pointers as having an address *plus some hidden
+// metadata*.
+//
+// This is to account for provenance
+// (https://doc.rust-lang.org/std/ptr/index.html#using-strict-provenance) and
+// wide pointers. See e.g.
+// https://doc.rust-lang.org/std/primitive.pointer.html#method.is_null: "unsized
+// types have many possible null pointers, as only the raw data pointer is
+// considered, not their length, vtable, etc. Therefore, two pointers that are
+// null may still not compare equal to each other."
+
+pub trait PointerExt<T: ?Sized>: Sized {
```

Comment:
This trait should be exported in `base_prelude` so that it can be used without importing std::pointer.

Reply:
done